Sources/drawing.py

```diff
@@ -0,0 +1,418 @@
+            if data.Sections.Count > 0:
+                self.draw_Sections(data.Sections)
+            if data.Observations.Count > 0:
+                self.draw_Observations(data.Observations)
+            if data.NodeInspections.Count > 0:
+                self.draw_NodeInspections(data.NodeInspections)
+            if data.NodeObservations.Count > 0:
+                self.draw_NodeObservations(data.NodeObservations)
+            if data.Inspections.Count > 0:
+                self.draw_Inspections(data.Inspections)
+                
+        canvas = self.qgis.mapCanvas()
+        canvas.zoomToFullExtent()
+                
+    @pyqtSlot(int, str)       
+    def select_feature(self, entity_type, feature_id):
+        if self.VX.IsConnected and self.layers_created:
+            layer = None
+            if entity_type == 0:
+                layer = self.created_layers[1]
+            elif entity_type == 3:
+                layer = self.created_layers[2]
+            if layer != None:
+                mc = self.qgis.mapCanvas()
+                for l in mc.layers():
+                    if l.type() == l.VectorLayer:
+                        l.removeSelection()
+                request = QgsFeatureRequest().setSubsetOfAttributes(["OBJ_PK"], layer.fields()).setFilterExpression('"OBJ_PK"=\'%s\'' % feature_id)
+                features = layer.getFeatures(request)
+
+                for f in features:
+                    layer.select(f.id())
+
+                box = layer.boundingBoxOfSelected()
+                mc.setExtent(box)
+                mc.refresh()
+                
+    @pyqtSlot(list)
+    def delete_entities(self, entities):
+        sections = [entity for entity in entities if entity.EntityType == 0]
+        nodes = [entity for entity in entities if entity.EntityType == 3]
+        inspections = [entity for entity in entities if entity.EntityType == 1]
+        observations = [entity for entity in entities if entity.EntityType == 2]
+        nodeInspections = [entity for entity in entities if entity.EntityType == 5]
+        nodeObservations = [entity for entity in entities if entity.EntityType == 6]
+
+        self.delete_features(sections, self.created_layers[1])
+        self.delete_features(nodes, self.created_layers[2])
+        self.delete_features(inspections, self.created_layers[0])
+        self.delete_features(observations, self.created_layers[5])
+        self.delete_features(nodeInspections, self.created_layers[3])
+        self.delete_features(nodeObservations, self.created_layers[4])
+        self.qgis.mapCanvas().refresh()
+                                       
+    def get_selected(self):
+        count = 0
+        selected_shapes = []
+        layer = self.qgis.activeLayer()
+        if type(layer) is not type(None):
+            for feature in layer.selectedFeatures():
+                count += 1
+                selected_shapes.append(feature)
+        return count, selected_shapes, layer
```

Comment:
so selected_shapes is the same as layer.selectedFeatures() and count is len(selected_shapes), this is so redundant

Reply:
Changed